src/lib.rs

```diff
@@ -505,61 +596,131 @@ pub struct Category<BlockNumber, Moment, AccountId> {
+        pub ReactionByPost get(reaction_by_post) config(): double_map PostId, blake2_256(ForumUserId) => PostReaction;
+
+        /// Description for each item in a poll.
+        pub PollDesc get(poll_desc) config(): double_map ThreadId, blake2_256(u8) => Vec<u8>;
+
+        /// Each account's data record to a poll.
+        pub PollDataByAccount get(poll_by_account) config(): double_map ThreadId, blake2_256(ForumUserId) => PollData;
+
+        /// Poll statistics
+        pub PollStatistics get(poll_statistics) config(): double_map ThreadId, blake2_256(PollData) => u64;
+
+        /// Input constraints for description text of category title.
         pub CategoryTitleConstraint get(category_title_constraint) config(): InputValidationLengthConstraint;
+
+        /// Input constraints for description text of category description.
         pub CategoryDescriptionConstraint get(category_description_constraint) config(): InputValidationLengthConstraint;
+
+        /// Input constraints for description text of thread title.
         pub ThreadTitleConstraint get(thread_title_constraint) config(): InputValidationLengthConstraint;
+
+        /// Input constraints for description text of post.
         pub PostTextConstraint get(post_text_constraint) config(): InputValidationLengthConstraint;
+
+        /// Input constraints for description text of moderation thread rationale.
         pub ThreadModerationRationaleConstraint get(thread_moderation_rationale_constraint) config(): InputValidationLengthConstraint;
+
+        /// Input constraints for description text of moderation post rationale.
         pub PostModerationRationaleConstraint get(post_moderation_rationale_constraint) config(): InputValidationLengthConstraint;
+
+        /// Input constraints for label name.
+        pub LabelNameConstraint get(label_name_constraint) config(): InputValidationLengthConstraint;
+
+        /// Input constraints for description text of each item in poll.
+        pub PollDescConstraint get(poll_desc_constraint) config(): InputValidationLengthConstraint;
+
+        /// Input constraints for number of items in poll.
+        pub PollItemsConstraint get(poll_items_constraint) config(): InputValidationLengthConstraint;
+
+        /// Input constraints for user name.
+        pub UserNameConstraint get(user_name_constraint) config(): InputValidationLengthConstraint;
+
+        /// Input constraints for user introduction.
+        pub UserSelfIntroductionConstraint get(user_self_introduction_constraint) config(): InputValidationLengthConstraint;
+
+        /// Labels could be applied to category and thread
+        pub CategoryThreadLabels get(category_thread_labes) config(): map LabelId => Vec<u8>;
+
+        /// Next label identifier
+        NextLabelId get(next_label_id) config(): u64;
+
+        /// All labels applied to a category
+        CategoryLabels get(category_labels) config(): map CategoryId => Vec<LabelId>;
```

Comment:
Accepted. Considering the label text is short. I will store it directly instead of index of each label. Like you mentioned, storage size is not critical for now.